app/processing/parse-payment-file.js

```diff
@@ -1,16 +1,19 @@
 const getPaymentRequestsFromFile = require('./get-payment-requests-from-file')
-const { sendBatchProcessedEvents } = require('../event')
+const { sendBatchProcessedEvents, sendPaymentRequestInvalidEvents } = require('../event')
 const { sendPaymentBatchMessage } = require('../messaging')
 
 const parsePaymentFile = async (filename, fileBuffer, scheme, sequence) => {
   try {
-    const { paymentRequests, batchExportDate } = await getPaymentRequestsFromFile(fileBuffer, scheme)
-    await sendBatchProcessedEvents(filename, paymentRequests, sequence, batchExportDate)
-    await sendPaymentBatchMessage(paymentRequests)
+    const { paymentRequestsCollection, batchExportDate } = await getPaymentRequestsFromFile(fileBuffer, scheme)
+
+    await sendBatchProcessedEvents(paymentRequestsCollection.successfulPaymentRequests, filename, sequence, batchExportDate)
```

Comment:
This throws an error if the only payment request in the file was invalid as we're trying to publish an empty array of messages.

Could we add a conditional here so we only try to send messages if we have messages to send and we only send failures if we have failures to send?

Reply:
Aaaahhh really? No way

I added a conditional inside `sendBatchProcessedEvents` which:

1. check if `paymentRequestsCollection.successfulPaymentRequests` has `.length` (an empty array should be caught here and ignored)
2. try to run `.map` on it (this eliminates issues if a string or object or other data types got thro `.length` check